_posts/2016-05-03-podspec-checksums.markdown

```diff
@@ -0,0 +1,74 @@
+target 'MyApp'
+```
+
+A developer running `pod install` would get the latest `3.x` version, which could be `3.1` originally, but then 6 months later they could get `3.4` - without a lockfile there is no way to keep track of the specific build. This is why it should always be in your code repo. In the case above my lockfile looks like this:
+
+``` yaml
+PODS:
+  - AFNetworking/Serialization (3.1.0)
+
+DEPENDENCIES:
+  - AFNetworking/Serialization (~> 3.0)
+
+SPEC CHECKSUMS:
+  AFNetworking: 5e0e199f73d8626b11e79750991f5d173d1f8b67
+
+PODFILE CHECKSUM: 876ceaa409f4ade2b3d58d310dbe026393824bcc
+
+COCOAPODS: 1.0.0.beta.8
+````
+
+### What do the Spec Checksums do?
+
+With the CocoaPods Master Specs repo, we do our best [to try](https://github.com/CocoaPods/Specs/pull/12199) and ensure a write-once repository of Podspecs for the public. However, there are many times when you cannot guarantee that every you have the same version of a Podspec as everyone else in your team.
+
+So, CocoaPods makes a checksum of the JSON representation of your Podspec and keeps that in the lockfile. You can easily [replicate](https://github.com/CocoaPods/CocoaPods/issues/3371) the work to generate a checksum with:
+
+``` sh
+~/D/MyApp ⏛  pod ipc spec ~/.cocoapods/repos/master/Specs/AFNetworking/3.1.0/AFNetworking.podspec.json  | openssl sha1
+5e0e199f73d8626b11e79750991f5d173d1f8b67
+```
+
+### So why am I seeing churn?
+
+A normal git development flow when working with libraries is to:
+
+* Fork a library, change your Podfile to reflect that change
+* Make some changes
+* Commit them back to the main repo
+* Update the Podspec, then make changes bringing your Podfile back to a real (tagged) release
+
+CocoaPods is smart about updating your libraries behind the scenes, but it's not perfect. In order to avoid re-creating your entire Pods folder every time it will check whether your libraries are at the expected version and skip re-creating the whole process.
+
+In the example above, we used the CocoaPods' Specs repo version of the Podspec. In forked repos, e,g,
+
+``` ruby
+pod 'AFNetworking/Serialization', :git => "https://github.com/orta/AFNetworking.git", :commit => "6f31b5c7bcbd59d4dac7e92e215d3c2c22f3400e"
+```
+
+The Podspec is saved into the `Pods` directory in JSON format at `Pods/Local\ Podspecs/AFNetworking.podspec.json`, this is to ensure there's always access within the CocoaPods sandbox for the Podspecs, and speed probably. This is the podspec used for generating the checksum.
+
+Effectively, during development you would have used `pod update [library]` to update your library you were working, then made changes to a podspec without a final `pod update [library]`. From that point on you would only use `pod install` which considers the library to not need updating. So then when others pull down your changes, their Podspecs are a different sha, and CocoaPods will keep overwriting the version in the lockfile without forcing an update.
```

Comment:
“you were working **on**”

I don’t really follow what’s being explained here, though, can you rephrase it?